### ISSUE TITLE
Plugins not loading on windows when path include space characters

### ISSUE THREAD

**Issue by casz** [hashicorp/packer#8413]
#### Overview of the Issue

Our IT department is evil and have introduced spaces in our usernames 🙈.
seems like packer cannot load plugins from appdata folder:

```
ll 'C:\Users\Joseph Petersen\AppData\Roaming\packer.d\plugins'
-a----       02-08-2019     10:44    44,13MB packer-builder-vsphere-clone.exe
-a----       28-10-2019     16:16    34,95MB packer-builder-vsphere-iso.exe
-a----       06-05-2019     22:53    29,39MB packer-provisioner-windows-update.exe
```

At least that seems to be the case for packer-provisioner-windows-update.exe

If I place the packer-provisioner-windows-update.exe in the current directory where I am executing packer from everything is fine. `C:\git\work\ThreeShape.Baseboxes`

#### Reproduction Steps

Create a user profile with spaces

### Packer version

```
packer version
Packer v1.4.5
```

### Operating system and Environment details

Windows 10 1909
Also a confirmed to be an issue on 1903

**Comment by SwampDragons** [comment]
Oh no! Bet we're not the only project that username policy has caused some havoc with.  😄. We'll try to take a look at this before the 1.5.0 release; thanks for reaching out.

**Comment by casz** [comment]
it seems to load builders just fine, but has trouble with provisioners perhaps.
I must admit I am not sure how to debug plugin loading.

**Comment by nywilken** [comment]
Hi @casz thanks again for opening up this issue. I was taking a look at Packer 1.4.5 and 1.5.1 to see if I could reproduce this issue. By placing two custom provisioners into the default plugin directory using a username which contains spaces. But saw that plugins were at least being discovered by Packer. 


```
C:\Users\Packerista Jones\Downloads\bin> set PACKER_LOG=1
C:\Users\Packerista Jones\Downloads\bin>packer build build.json
2020/01/17 13:03:12 [INFO] Packer version: 1.4.5
2020/01/17 13:03:12 Packer Target OS/Arch: windows amd64
2020/01/17 13:03:12 Built with Go Version: go1.13.4
2020/01/17 13:03:12 Detected home directory from env var: C:\Users\Packerista Jones\AppData\Roaming
2020/01/17 13:03:12 [DEBUG] Discovered plugin: comment = C:\Users\Packerista Jones\AppData\Roaming\packer.d\plugins\packer-provisioner-comment.exe
2020/01/17 13:03:12 [DEBUG] Discovered plugin: windows-update = C:\Users\Packerista Jones\AppData\Roaming\packer.d\plugins\packer-provisioner-windows-update.exe
```

In running a simple build configuration I see the Windows update plugin is at least trying to execute. I'm using the null builder here so I expect the upload part to fail. 
```
C:\Users\Packerista Jones\Downloads\bin>packer build build.json
null output will be in this color.

==> null: Running local shell script: C:\Users\PACKER~1\AppData\Local\Temp\packer-shell035597327.cmd
    null:
    null: C:\Users\Packerista Jones\Downloads\bin>echo simple shell
    null: simple shell
==> null: Uploading the Windows update elevated script...
==> null: Provisioning step had errors: Running the cleanup provisioner, if present...
Build 'null' errored: Upload is not implemented when communicator = 'none'

==> Some builds didn't complete successfully and had errors:
--> null: Upload is not implemented when communicator = 'none'

==> Builds finished but no artifacts were created.
```

To help debug the loading  of the plugin I would suggest enabling the [Packer debug](https://www.packer.io/docs/other/debugging.html#debugging-packer-in-powershell-windows) and look to see if the plugins are being loaded (top of the file). Do you know if the custom provisioner has the executable bit set? I believe the file will not load if it is not executable. 

 If you can upload a redacted copy of your debug log to this issue via a gist so that we can help take a look at what might be happening as well. A redacted version of your build configuration for reproducing the problem would also be helpful. Look forward to hearing back

<details>
<summary>Contents of plugin directory</summary>

```
C:\Users\Packerista Jones\AppData\Roaming>dir packer.d\plugins
 Volume in drive C has no label.
 Volume Serial Number is 2061-6B90

 Directory of C:\Users\Packerista Jones\AppData\Roaming\packer.d\plugins

01/17/2020  01:13 PM    <DIR>          .
01/17/2020  01:13 PM    <DIR>          ..
01/17/2020  12:59 PM        29,037,056 packer-provisioner-comment.exe
01/17/2020  01:13 PM        31,587,328 packer-provisioner-windows-update.exe
               2 File(s)     60,624,384 bytes
```
</details>

**Comment by nywilken** [comment]
Hi @casz just wanted to follow up to see if you are still running into this issue with paths that contain spaces. Since opening this bug there has been some work done around support for a PACKER_PLUGIN_PATH that can be used for changing the location of where plugins, not the config, should be sourced from so this may help if spaces or other chars are causing plugin loading issues. 

In cases where the plugin does not end in an `.exe` extension or not executable Packer will skip loading the plugin but there is usually a message in the log indicating that it was skipped. To enable the Packer debug logs you can set the environment variable `PACKER_LOG=1`; setting `PACKER_LOG_PATH=<some file>` will save the logs to a file for sharing. 

You can give the PACKER_PLUGIN_PATH env setting a run by downloading a nightly build of Packer at https://github.com/hashicorp/packer/releases/tag/nightly

If you are no longer running into this issue please feel free to close it.

**Comment by casz** [comment]
I'll try and look into tomorrow 😅

**Comment by nywilken** [comment]
Hi @casz thanks for taking a look at the latest Packer release.

Seeing as I am unable to reproduce this issue and that support for the PACKER_PLUGIN_PATH environment variable is set to be released in 1.5.2, later this week. I am closing this issue. If after additional testing with 1.5.1 or 1.5.2 (once released) you find that you are still having issues. Please open a new issue with a redacted copy of the build logs `PACKER_LOG=1 PACKER_LOG_PATH=debug.log packer build build.json` and we will try to help figure out what might be happening. 

Cheers

**Comment by casz** [comment]
Thanks for adding the env
